clang/docs/SYCLFERFCLoweringKernel.md

```diff
@@ -0,0 +1,88 @@
+# SYCL Kernel
+
+A SYCL construct such as 'single_task' takes a function object as one of its arguments.   The contents of this function object are executed on the device.   To enable this, the function object is converted into the format of an OpenCL kernel.
```

Comment:
- People usually knows about `parallel_for` as it is present on any SYCL presentation slides, but `single_task` not so much.
- As this is a FE RFC only, it is not immediately obvious why this transformation is required

```suggestion
A SYCL construct such as `parallel_for` or `single_task` takes a function object as one of its arguments.
The contents of this function object are executed on the device.
However, as the SYCL runtime can rely on other offload APIs like OpenCL or CUDA to execute the function object, therefore it needs to respect the calling convention of these API.
To enable this, the function object is converted into the format of an OpenCL kernel.
```